### ISSUE TITLE
popup is not working in IE 

### ISSUE THREAD

**Issue by aline-almeida** [AzureAD/azure-activedirectory-library-for-js#544]
When using popup, after entering credentials, it doesn't close and app continues in the popup window instead of previous window.
See [this thread ](http://stackoverflow.com/questions/43449656/adal-js-with-popup-not-working-on-edge-internet-explorer)  for more details.

**Comment by rohitnarula7176** [comment]
In IE, can you go to Internet options->Security and uncheck the Enable protected mode property. This property when checked returns a null reference for the opened popup window and hence we cannot close it. Please let me know if that fixes your issue.

**Comment by aline-almeida** [comment]
Unfortunatelly, it didn't fix it...

**Comment by jmprieur** [comment]
@aline-almeida : could you please have a look at https://github.com/AzureAD/microsoft-authentication-library-for-js/wiki/Using-msal.js-with-Internet-Explorer (which is for msal.js, but there are common tips) and tell us if this worked?

**Comment by aline-almeida** [comment]
@jmprieur : I've added the script line and also disabled protected mode... problem remains.

**Comment by cserkaran** [comment]
Is there any solution to this issue ?

**Comment by jmprieur** [comment]
@aline-almeida @cserkaran did you try with ADAL.js 1.0.15?

**Comment by CivilianZero** [comment]
I am also encountering this issue, but only since moving to Angular 4+. Any solutions so far?

**Comment by aline-almeida** [comment]
I've now moved to Angular 4 and adal-angular 1.0.15 following this post: 
[https://blogs.msdn.microsoft.com/premier_developer/2017/04/26/using-adal-with-angular2/](https://blogs.msdn.microsoft.com/premier_developer/2017/04/26/using-adal-with-angular2/)
I can successfully login and catch the login callback without the popup. That's fine with me.

**Comment by rohitnarula7176** [comment]
@aline-almeida Thank you for your input. Closing this issue.

**Comment by bondort** [comment]
I still have this issue in angularJs and I can't upgrade my project to angular-4...
I think this issue needs to stay open, as it was never resolved, the OP just moved to a different angular version, which is not a fix.

Should I just tell my client that IE is not supported?
This works perfectly in Chrome, Firefox, and even Edge...

Thanks

**Comment by bondort** [comment]
In case anyone has this problem, I managed to find a simple solution.
I've debugged the login process with one of my colleges, and we found out that the issue is in the _getHash function, as IE sends a bigger part of the URL segment.
Other browsers only had the '#PARAMETERS', but IE had '#!/URL_SEGMENTS#PARAMETERS'.

so TL:DR the fix was to add a simple regex, to remove the extra URL segments from IE:
`hash = hash.replace(/(#!\/)(.*)(#)/g, "#");`
Add this to the first place in the _getHash function, and IE will work in angularJs
